CHANGES.md

```diff
@@ -1,3 +1,10 @@
+v2.0.0 / 2023-08-02
+===================
+
+  * Complete overhaul of implementation (taken from Rematch2.jl)
+  * Code generation via an optimized decision automaton.
+  * Require Julia 1.5
+  * Note incompatibility: drops support for multidimensional arrays.
```

Comment:
maybe add a "see something for migration guide"? or breaking changes or whatever needs to be called out?